packages/api/models/granules.js

```diff
@@ -471,6 +478,111 @@ class Granule extends Manager {
     return Promise.all(granules.Items.map((granule) =>
       super.delete({ granuleId: granule.granuleId })));
   }
+
+  /**
+   * Get the set of fields which are mutable based on the granule status.
+   *
+   * @param {Object} record - A granule record
+   * @returns {Array} - The array of mutable field names
+   */
+  _getMutableFieldNames(record) {
+    if (record.status === 'running') {
+      return ['createdAt', 'updatedAt', 'timestamp', 'status', 'execution'];
+    }
+    return Object.keys(record);
```

Comment:
Does this mean that, for a record with a "completed" status, we can overwrite `createdAt`? Also, does this mean that `status` can be overwritten for by a "running" record?

Reply:
For a `completed` or `failed` record, you can overwrite anything, which this test is meant to show:

https://github.com/nasa/cumulus/blob/CUMULUS-1717-update-granule-reporting/packages/api/tests/models/granules/test-granule-reporting.js#L63

For a `running` record, you can update the `status` **but only when the execution has changed**:

https://github.com/nasa/cumulus/blob/CUMULUS-1717-update-granule-reporting/packages/api/models/granules.js#L562

This allows the scenario where a granule is being re-ingested or the execution re-run from the AWS console and we want the granule to go from "completed" to "running", but the execution should also have changed.

Reply:
So if a "completed" event is processed first, then a "running" event is processed for the same execution, the "running" event can make no updates to the record.

If a record is in the "completed" state, then a "running" event is received from a different execution, the fields listed on line 490 can be updated. In that case, do we want to be updating the "createdAt" field?

Reply:
> So if a "completed" event is processed first, then a "running" event is processed for the same execution, the "running" event can make no updates to the record.

Correct

> If a record is in the "completed" state, then a "running" event is received from a different execution, the fields listed on line 490 can be updated.

Correct. I'm not sure about `createdAt` in that case. It is normally set from the workflow_start_time, and that value would be different from the new execution. But I guess we may not want to change it so that the time that the record was originally `createdAt` is preserved. So - maybe not?